engine/src/engine.rs

```diff
@@ -280,27 +531,67 @@ impl<'a> ModuleImportResolver for RuntimeModuleImportResolver {
     }
 }
 
-pub fn exec<T: TrackingCopy, G: GlobalState<T>>(
-    module: wasmi::Module,
-    account_addr: [u8; 20],
-    gs: &G,
-) -> Result<ExecutionEffect, Error> {
+fn instance_and_memory(parity_module: Module) -> Result<(ModuleRef, MemoryRef), Error> {
+    let module = wasmi::Module::from_parity_wasm_module(parity_module)?;
     let resolver = RuntimeModuleImportResolver::new();
     let mut imports = ImportsBuilder::new();
     imports.push_resolver("env", &resolver);
     let instance = ModuleInstance::new(&module, &imports)?.assert_no_start();
 
     let memory = resolver.mem_ref()?;
+    Ok((instance, memory))
+}
+
+fn sub_call<T: TrackingCopy>(
+    parity_module: Module,
+    args: Vec<Vec<u8>>,
+    current_runtime: &mut Runtime<T>,
+) -> Result<Vec<u8>, Error> {
+    let (instance, memory) = instance_and_memory(parity_module.clone())?;
+    let known_urefs: HashSet<Key> = HashSet::new();
+    let mut runtime = Runtime {
+        args,
+        memory,
+        state: current_runtime.state,
+        known_urefs,
+        module: parity_module,
+        result: Vec::new(),
+    };
+
+    let result = instance.invoke_export("call", &[], &mut runtime);
```

Comment:
Why are you not passing arguments as an argument to `invoke_export` method? The doc says:
> This function finds exported function by a name, and calls it with provided arguments and external state

Reply:
Is there some other `invoke_export` I am not seeing? In the body I find:
```rust
let extern_val = self.export_by_name(func_name).ok_or_else(|| {                                                                                                  
  Error::Function(format!("Module doesn't have export {}", func_name))                                                                                         
})?;
```

This means that for exported methods we don't have to rename their function names to `call`.

Reply:
We're not passing arguments directly because wasm only supports (single) numbers, so it would not work to pass in a higher level data structure. This is why I have the `get_arg` mechanism instead.

I know there is nothing special about the name `call`, but I thought it would make things cleaner if the entry point to a contract always had the same name. Like how the entry point to a regular program is always called `main`. Otherwise we would need to store somewhere additional information about what the name of the entry point export is.